src/main/resources/com/github/checkstyle/regression/extract/ExtractInfoGeneratorTest.java

```diff
@@ -0,0 +1,83 @@
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.puppycrawl.tools.checkstyle.internal.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.ModuleReflectionUtils;
+
+/**
+ * This file would be injected into checkstyle project and invoked by maven command
+ * to generate the module extract info file.
+ * @author LuoLiangchen
+ */
+public class ExtractInfoGeneratorTest {
+    /**
+     * Generates the extract info file named as "checkstyle_modules.json".
+     * @throws IOException failure when generating the file
+     */
+    @Test
+    public void generateExtractInfoFile() throws IOException {
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        final List<JsonObject> objects = CheckUtil.getCheckstyleModules().stream()
```

Comment:
One issue with `CheckUtil.getCheckstyleModules` is we can't test on old commits.
I think we should find someway to be more backward compatible. This can be a new issue.